### ISSUE TITLE
TypeScript modules fail

### ISSUE THREAD

**Issue by SetTrend** [Microsoft/vscode#1803]
Whenever I try to export some code in TypeScript I get an error message claiming that VS Code `Cannot compile modules unless the --module flag is provided.`:

![typescript module](https://cloud.githubusercontent.com/assets/9283914/12129513/6840e888-b404-11e5-9732-f8b231e70aaa.png)

**Comment by dbaeumer** [comment]
You need to tell the TypeScript language service that you want to use external modules. By default TypeScript defaults to internal modules. The module system used by TypeScript can be controlled by the module property in the tsconfig.json file. For example
```json
{
    "compilerOptions": {
        "target": "ES5",
        "module": "commonjs"
    }
}
```
to tell TS to use node's module system.

**Comment by SetTrend** [comment]
Perfect! Thanks :clap:
<br/>
However, there still seems to be a problem when the class name begins with an underscore:

![baseerror](https://cloud.githubusercontent.com/assets/9283914/12157699/e7330fca-b4d3-11e5-9d23-f2ba719138b8.png)

I have [set up a repository with my sample project](https://github.com/SetTrend/TS18) for another issue I observe with TypeScript (Microsoft/Typescript#6382). Perhaps you may want to take some files and my setup from there?

**Comment by dbaeumer** [comment]
Do you see the same problem when compiling using tsc on the command line. If so pls report against the Microsoft/TypeScript repository.

**Comment by SetTrend** [comment]
I have now followed the instructions from the other issue and now I'm able to compile flawlessly.

However, VS Code still shows the same two (non existing) errors:

![codeerror](https://cloud.githubusercontent.com/assets/9283914/12187769/0b0f4a5e-b5af-11e5-86d1-d7cb39fd494c.png)

<br/>
I have uploaded the updated sample to the repository. Perhaps you want to clone it to be able to reproduce the issue?

Perhaps the TSC version coming with VS Code might be obsolete compared to the TSC version I'm using for my build? I'm using TypeScript v1.8 for my build.

If that'd be the case, where would an updated TSC version for VS Code need to be saved to so VS code will use the new one? Can VS Code deal with newer TSC versions?
